app/src/main/java/com/example/fitappa/Profile/Profile.java

```diff
@@ -132,6 +132,38 @@ public void setUserExtraInfo(String weight, String height, String firstName, Str
         user.setWeight(weight);
     }
 
+    /**
+     * gets a string of the users weight
+     * @return returns string of their weight
+     */
+    public String getUserWeight(){
```

Comment:
change to retrieveUserWeight because firebase will take any method that starts with a get and put it in the database so we gotta work around it :(

and I think you can remove public here unless this method is being accessed outside the package

Reply:
actually if you remove public it shouldn't be accessed by firebase anymore so you can leave the name of the method the same (with get)

If the method needs to be public (if it's accessed by something outside of the package) then you will need to change the name